molecule/mysql_hardening/converge.yml

```diff
@@ -24,6 +24,19 @@
         - mysql_python_package_debian is not defined
         - ansible_distribution != "Ubuntu"
         - ansible_distribution_major_version|int < 20
+
+    - name: Use Python 3 on Suse
+      set_fact:
+        ansible_python_interpreter: /usr/bin/python3
+      when:
+        - ansible_os_family == 'Suse'
+
+    - name: install required python packages on Suse
+      ansible.builtin.shell: zypper -n install python-xml python3-rpm python3-PyMySQL
+      args:
+        removes: /usr/bin/zypper
```

Comment:
shouldn't this be a prepare task?

Also using `removes` here seems like a very ugly workaround, why not use the fact `ansible_os_family` like in the other tasks?

Reply:
ansible.builtin.shell and ansible.builtin.command may not take when: Hence I had to resort to ugly workarounds...